enry.h

```diff
@@ -0,0 +1,277 @@
+}
+
+std::tuple<std::string, bool> GetLanguageByShebang(std::string_view content)
+{
+    GetLanguageByShebang_return go_return = GetLanguageByShebang(to_GoByteSlice(content));
+    return std::make_tuple(std::string(go_return.r0.p, go_return.r0.n), bool(go_return.r1));
+}
+
+std::tuple<std::string, bool> GetLanguageByVimModeline(std::string_view content)
+{
+    GetLanguageByVimModeline_return go_return = GetLanguageByVimModeline(to_GoByteSlice(content));
+    return std::make_tuple(std::string(go_return.r0.p, go_return.r0.n), bool(go_return.r1));
+}
+
+std::vector<std::string> GetLanguageExtensions(std::string_view language)
+{
+    GoSlice go_slice{};
+    GetLanguageExtensions(to_GoString(language), &go_slice);
+    return from_GoStringSlice(go_slice);
+}
+
+std::vector<std::string> GetLanguages(std::string_view filename, std::string_view content)
+{
+    GoSlice go_slice{};
+    GetLanguages(to_GoString(filename), to_GoByteSlice(content), &go_slice);
+    return from_GoStringSlice(go_slice);
+}
+
+std::vector<std::string> GetLanguagesByContent(std::string_view filename, std::string_view content, const std::initializer_list<std::string>& candidates = {})
+{
+    GoSlice go_candidates = to_GoStringSlice(candidates);
+    GoSlice go_slice{};
+    GetLanguagesByContent(to_GoString(filename), to_GoByteSlice(content), go_candidates, &go_slice);
+    std::vector<std::string> vec = from_GoStringSlice(go_slice);
+
+    free(go_candidates.data);
+    go_candidates.data = nullptr;
+    return vec;
+}
+
+std::vector<std::string> GetLanguagesByEmacsModeline(std::string_view filename, std::string_view content, const std::initializer_list<std::string>& candidates = {})
+{
+    GoSlice go_candidates = to_GoStringSlice(candidates);
+    GoSlice go_slice{};
+    GetLanguagesByEmacsModeline(to_GoString(filename), to_GoByteSlice(content), go_candidates, &go_slice);
+    std::vector<std::string> vec = from_GoStringSlice(go_slice);
+
+    free(go_candidates.data);
+    go_candidates.data = nullptr;
+    return vec;
+}
+
+std::vector<std::string> GetLanguagesByExtension(std::string_view filename, std::string_view content, const std::initializer_list<std::string>& candidates = {})
+{
+    GoSlice go_candidates = to_GoStringSlice(candidates);
+    GoSlice go_slice{};
+    GetLanguagesByExtension(to_GoString(filename), to_GoByteSlice(content), go_candidates, &go_slice);
+    std::vector<std::string> vec = from_GoStringSlice(go_slice);
+
+    free(go_candidates.data);
+    go_candidates.data = nullptr;
+    return vec;
+}
+
+std::vector<std::string> GetLanguagesByFilename(std::string_view filename, std::string_view content, const std::initializer_list<std::string>& candidates = {})
+{
+    GoSlice go_candidates = to_GoStringSlice(candidates);
+    GoSlice go_slice{};
+    GetLanguagesByFilename(to_GoString(filename), to_GoByteSlice(content), go_candidates, &go_slice);
+    std::vector<std::string> vec = from_GoStringSlice(go_slice);
+
+    free(go_candidates.data);
```

Comment:
Manual memory management kind-of reminds me that it would be nice to have a CI job that runs a loop over some sample data \w something like https://clang.llvm.org/docs/LeakSanitizer.html, in order to avoid situations like https://github.com/go-enry/go-enry/issues/36 😁

Reply:
Sure thing.
I've already tested with `valgrind` and `ASan`.  I'll add them to CI  in the next step.